src/theory/arith/arith_rewriter.cpp

```diff
@@ -381,6 +383,21 @@ RewriteResponse ArithRewriter::postRewriteMult(TNode t){
   return RewriteResponse(REWRITE_DONE, res.getNode());
 }
 
+RewriteResponse ArithRewriter::postRewritePow2(TNode t)
+{
+  Assert(t.getKind() == kind::POW2);
+  NodeManager* nm = NodeManager::currentNM();
+  // if constant, we eliminate
+  if (t[0].isConst())
+  {
+    Integer i = t[0].getConst<Rational>().getNumerator();
```

Comment:
What if it is rational?

Reply:
Then it should fail on type checking, [here](https://github.com/cvc5/cvc5/pull/6756/files#diff-c38305abf5d7b3f2e1ad0c47b5525b14721ba385967d30012a9cfcac0cc1ae5bR146) no? Anyway, I've added an assertion in 84b524d.